Epona/include/Epona/Debug.hpp

```diff
@@ -0,0 +1,60 @@
+/*
+* Copyright (C) 2017 by Godlike
+* This code is licensed under the MIT license (MIT)
+* (http://opensource.org/licenses/MIT)
+*/
+#ifndef EPONA_DEBUG_HPP
+#define EPONA_DEBUG_HPP
+
+#include <functional>
+#include <vector>
+#include <glm/glm.hpp>
+
+namespace epona {
+    template <typename T>
+    class QuickhullConvexHull;
+} // namespace epona
+
+namespace epona
+{
+namespace debug
+{
+
+/* Singleton class that provides callback interface for debugging */
+class Debug
+{
+public:
+    Debug() = default;
+    Debug& operator==(Debug const&) = delete;
+    Debug(Debug&) = delete;
+    Debug& operator==(Debug&&) = delete;
+    Debug(Debug&&) = delete;
+
+    /* Returns reference to the singleton instance of the class */
+    static Debug& GetInstace();
+
+    /**
+     * @brief QuickhullConvexHull debug call function
+     *
+     * This method is called from within the QuickhullConvexHull class
+     * during the calculation. It than proxies all the arguments to the currently
+     * set callback functor.
+     *
+     * @param[in] convexHull   current convex hull instance
+     * @param[in] vertexBuffer source vertices for the convex hull
+     */
+    static void QuickhullConvexHullCall(
+        QuickhullConvexHull<std::vector<glm::dvec3>>& convexHull,
+        std::vector<glm::dvec3>& vertexBuffer
+    );
+
+    /* Callback for the QuickhullConvexHull debug calls */
+    std::function<
```

Comment:
can this be static as well? this well there's no need to have this class as singleton at all